wrappers/node/src/api/connection.ts

```diff
@@ -0,0 +1,174 @@
+import * as ffiNapi from '@hyperledger/vcx-napi-rs';
+import { VCXInternalError } from '../errors';
+import { ISerializedData, ConnectionStateType } from './common';
+import { VcxBaseWithState } from './vcx-base-with-state';
+import { IPwInfo } from './utils';
+
+export type INonmediatedConnectionInvite = string;
+
+export type INonmeditatedConnectionData = string;
+
+export interface IEndpointInfo {
+  serviceEndpoint: string,
+  routingKeys: string[],
+}
+
+export class NonmediatedConnection extends VcxBaseWithState<INonmeditatedConnectionData, ConnectionStateType> {
```

Comment:
I guess going forward we can remove existing `Connection` and rename `NonmediatedConnection -> Connection`, agree?

Reply:
Yes - I decided to go for fairly consistent `NomediatedConnection` naming convention from to wrapper layer for purposes of backwards-compatibility, but also to make it easy to rename in the future once the old, mediated connection handler is gone.